pkg/coreapi/apiutil/apiutil.go

```diff
@@ -0,0 +1,54 @@
+package apiutil
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/inngest/inngest/pkg/enums"
+	"github.com/inngest/inngest/pkg/execution/state"
+	"github.com/inngest/inngest/pkg/publicerr"
+	"github.com/oklog/ulid/v2"
+)
+
+var (
+	ErrRunIDInvalid = fmt.Errorf("The run ID specified is invalid")
+)
+
+// EventAPIResponse is the API response sent when responding to incoming events.
+type EventAPIResponse struct {
+	IDs    []string `json:"ids"`
+	Status int      `json:"status"`
+	Error  error    `json:"error,omitempty"`
+}
+
+// CancelRun cancels a run for a given run ID, returning consistent errors for public APIs
+func CancelRun(ctx context.Context, sm state.Manager, runID ulid.ULID) error {
+	if sm == nil {
+		return fmt.Errorf("no state manager supplied to cancel run")
```

Comment:
does this error message consider an abstration leak?
in other words, if I get this error, what do I as the user can do to fix it?

Reply:
The idea is that we use this error to catch issues in integration tests, and that users _never_ see it.  It's an internal error, hence no `publicerr.Error`